Add driver for LTP8800-1A, LTP8800-4A and LTP8800-2

cencarna opened the pull request:
## PR Description

- This adds driver support and device tree bindings for LTP8800 uModule Regulator family.
- Datasheets [LTP8800-1A](https://www.analog.com/media/en/technical-documentation/data-sheets/ltp8800-1a.pdf), [LTP8800-4A](https://www.analog.com/media/en/technical-documentation/data-sheets/ltp8800-4a.pdf), [LTP8800-2](https://www.analog.com/media/en/technical-documentation/data-sheets/ltp8800-2.pdf)
- Tested on RPI4 using the following demo boards: [DC3190A-A](https://www.analog.com/media/en/technical-documentation/user-guides/dc3190a-a.pdf), [DC3190A-B](https://www.analog.com/media/en/technical-documentation/user-guides/dc3190a-b.pdf), [DC3190B-E](https://www.analog.com/media/en/technical-documentation/user-guides/dc3190b-e.pdf)

## PR Type
- [ ] Bug fix (a change that fixes an issue)
- [x] New feature (a change that adds new functionality)
- [ ] Breaking change (a change that affects other repos or cause CIs to fail)

## PR Checklist
- [x] I have conducted a self-review of my own code changes
- [x] I have tested the changes on the relevant hardware
- [ ] I have updated the documentation outside this repo accordingly (if there is the case)

nunojsa commented on drivers/hwmon/pmbus/ltp8800.c:
```diff
@@ -0,0 +1,75 @@
+		   PMBUS_HAVE_POUT,
+#if IS_ENABLED(CONFIG_SENSORS_LTP8800_REGULATOR)
+	.num_regulators = 1,
+	.reg_desc = ltp8800_reg_desc,
```
Since `pmbus_driver_info` is not `const` I would do it without the preprocessor as it looks neater to me... See below

nunojsa commented on drivers/hwmon/pmbus/ltp8800.c:
```diff
@@ -0,0 +1,75 @@
+	if (!i2c_check_functionality(client->adapter,
+				     I2C_FUNC_SMBUS_READ_BYTE_DATA |
+				     I2C_FUNC_SMBUS_READ_WORD_DATA))
+		return -ENODEV;
```
```
if (IS_ENABLED(CONFIG_SENSORS_LTP8800_REGULATOR)) {
    ltp8800_info.num_regulators = 1;
    ltp8800_info.ltp8800_reg_desc = ltp8800_reg_desc;
}

nunojsa commented on drivers/hwmon/pmbus/ltp8800.c:
```diff
@@ -0,0 +1,75 @@
+#include <linux/mod_devicetable.h>
+#include "pmbus.h"
+
+static const struct regulator_desc __maybe_unused ltp8800_reg_desc[] = {
```
Can you make sure we really need the `__mabe_unused`? I would expect the compiler to be smart enough to realize this is not (since CONFIG_SENSORS_LTP8800_REGULATOR is a compile time constant) used and remove it

nunojsa commented on Documentation/devicetree/bindings/hwmon/pmbus/adi,ltp8800.yaml:
```diff
@@ -0,0 +1,44 @@
+        hwmon@41 {
+            compatible = "adi,ltp8800-2";
+            reg = <0x41>;
+        };
```
I'm not sure if there's something against it these days... but this looks like it could go to:

https://elixir.bootlin.com/linux/v6.12-rc2/source/Documentation/devicetree/bindings/trivial-devices.yaml

nunojsa reviewed (commented):
Here it goes my 2 cents. Go ahead and send this upstream :)

nunojsa commented on MAINTAINERS:
```diff
@@ -12255,6 +12255,12 @@ S:	Maintained
+M:	Cedric Encarnacion <cedricjustine.encarnacion@analog.com>
+L:	linux-hwmon@vger.kernel.org
+S:	Maintained
+F:	Documentation/devicetree/bindings/hwmon/pmbus/adi,ltp8800.yaml
```
leftover...

cencarna commented:
v2:
- Moved dts bindings to trivial-devices.yaml
- Removed `IS_ENABLED(REGULATOR)` preprocessor. Regulator initialization moved to probe.
- Removed `__maybe_unused` attribute.
- housekeeping and build fixes

DT binding check always fails for some reason. Check complains this is due to title pattern which is already outside my changes.

nunojsa reviewed (approved):
LGTM,

Please send this upstream and let's merge it once it's accepted there.

nunojsa commented:
Status on this one?
